__tests__/index.spec.js

```diff
@@ -1,18 +1,253 @@
+    });
+
+    test('onTestResult: startTest, finishTest should be called two times with second parameter \'true\' if there are retries', () => {
+        const spyStartTest = jest.spyOn(reporter, '_startTest');
+        const spyFinishTest = jest.spyOn(reporter, '_finishTest');
+        testResult.testResults[0].invocations = 2;
+
+        reporter.onTestResult({}, testResult);
+
+        expect(spyStartTest).toHaveBeenCalledWith(testResult.testResults[0].title, true);
+        expect(spyFinishTest).toHaveBeenCalledWith(testResult.testResults[0], true);
+        expect(spyStartTest).toHaveBeenCalledTimes(2);
+        expect(spyFinishTest).toHaveBeenCalledTimes(2);
+    });
+
+    test('onRunComplete: finishLaunch should be called with tempLaunchId', () => {
+        reporter.tempLaunchId = 'tempLaunchId';
+
+        reporter.onRunComplete();
+
+        expect(reporter.client.finishLaunch).toHaveBeenCalledWith('tempLaunchId');
+    });
+
+    test('_startSuite: startTestItem should be called with parameters, tempSuiteId should be defined', () => {
+        const expectedStartTestItemParameter = {
+            type: 'SUITE',
+            name: 'suite name',
+            startTime: new Date().valueOf()
+        };
+        reporter.tempLaunchId = 'tempLaunchId';
+
+        reporter._startSuite('suite name');
+
+        expect(reporter.client.startTestItem).toHaveBeenCalledWith(expectedStartTestItemParameter, 'tempLaunchId');
+        expect(reporter.tempSuiteId).toEqual('startTestItem');
+    });
+
+    test('_startTest: startTestItem should be called with parameters, tempTestId should be defined', () => {
+        const expectedStartTestItemParameter = {
+            type: 'STEP',
+            name: 'test name',
+            retry: true,
+        };
+        reporter.tempLaunchId = 'tempLaunchId';
+        reporter.tempSuiteId = 'tempSuiteId';
+
+        reporter._startTest('test name', true);
+
+        expect(reporter.client.startTestItem).toHaveBeenCalledWith(expectedStartTestItemParameter, 'tempLaunchId', 'tempSuiteId');
+        expect(reporter.tempTestId).toEqual('startTestItem');
+    });
+
+    test('_sendLog: sendLog should be called with parameters', () => {
+        const expectedLogObjectParameter = {
+            message: 'message',
+            level: 'error',
+        };
+        reporter.tempTestId = 'tempTestId';
+
+        reporter._sendLog('message');
+
+        expect(reporter.client.sendLog).toHaveBeenCalledWith('tempTestId', expectedLogObjectParameter);
+    });
+
+    test('_finishTest: _finishPassedTest should be called if test status is passed', () => {
+        const spyFinishPassedTest = jest.spyOn(reporter, '_finishPassedTest');
+        const spyFinishFailedTest = jest.spyOn(reporter, '_finishFailedTest');
+        const spyFinishSkippedTest = jest.spyOn(reporter, '_finishSkippedTest');
+
+        reporter._finishTest({ status: testItemStatuses.PASSED, failureMessages: [] });
```

Comment:
Looks like this test might depend on previous tests (global state change). I don't think it's good.
The same for other tests below.

A unit test should be as much independend as it can be.

Reply:
Could you please clarify it "Looks like this test might depend on previous tests (global state change)"

Reply:
Sorry, misread the test. You just check what methods are called. In this case everything is good.